src/util/hass-translation.ts

```diff
@@ -36,9 +36,21 @@ export function findAvailableLanguage(language: string) {
     return LOCALE_LOOKUP[langLower];
   }
 
-  return Object.keys(translationMetadata.translations).find(
+  let translation = Object.keys(translationMetadata.translations).find(
     (lang) => lang.toLowerCase() === langLower
   );
+  if (translation) {
+    return translation;
+  }
+
+  if (language.includes("-")) {
+    translation = findAvailableLanguage(language.split("-")[0]);
+    if (translation) {
+      return translation;
+    }
```

Comment:
```suggestion
    return findAvailableLanguage(language.split("-")[0]);
```

Reply:
I assume that the next change is the preferred one?  I think this would create an infinite loop (I did try to figure out a way to end like this though :) )

Reply:
It will not create a loop, as it first checks if it has an `-`, the second run can never have a `-`.

Reply:
Sure, I see.  I was only looking at it in this context and forgetting the if around it.